errors.go

```diff
@@ -168,3 +168,10 @@ const (
 	ErrWrongSchemaVaersion           = 109 // Wrong schema version, current: %d, in request: %u
 	ErrSlabAllocMax                  = 110 // Failed to allocate %u bytes for tuple in the slab allocator: tuple is too large. Check 'slab_alloc_maximal' configuration option.
 )
+
+var (
+	ErrStrangerConn      = fmt.Errorf("the passed connected request doesn't belong to the current connection or connection pool")
+	ErrNilResponsePassed = fmt.Errorf("pased nil response")
+	ErrNilResponseData   = fmt.Errorf("response Data is nil")
+	ErrWrongDataFormat   = fmt.Errorf("response Data format is wrong")
+)
```

Comment:
I see the next advantages of having prepared errors:
- less allocations
- no need to compare strings of errors for user, if there is a logic that considers some errors
- easy to reuse
Disadvantages:
- a lot of errors could confuse the user
- there will be a risk to break the user's logic after renaming some error (so it will be a breaking change)

Reply:
>     * less allocations

You don't have to worry about it in case an error object creation. This is not the main scenario. We assume that it is unlikely.
 
>     * no need to compare strings of errors for user, if there is a logic that considers some errors

That's a good reason. But here we need a systematic approach. I.e., either all errors in this way or none. Otherwise, it will only confuse the user.

>    * easy to reuse

That's a good reason too. But between copy-paste strings between modules and add the strings into public API it is better to choose copy-paste.

I'm not against a systematic approach here. It would be great to declare all possible `go-tarantool` errors, maybe to think about naming and to sort it in some order. But I think it's out of scope of the issue.

Reply:
+ Easy to use it in tests.
Okay, let's leave it out of scope.